### DISCUSSION TITLE
Tailwindcss 2.0 darkmode

### DISCUSSION THREAD

**Discussion by nos-nart** [vercel/next.js#19343] [Help]
I'm trying to do [Tailwind darkmode](https://tailwindcss.com/docs/dark-mode) with Nextjs, following tailwind docs I have `tailwind.config.js` like this
```js
module.exports = {
  future: {
    removeDeprecatedGapUtilities: true,
    purgeLayersByDefault: true,
  },
  purge: false,
  darkMode: 'class',
  theme: {
    extend: {},
  },
  variants: {},
  plugins: [],
};
```
and file  `_document.js`
```js
import Document, { Html, Head, Main, NextScript } from 'next/document'
import storage from 'local-storage-fallback';
import React from 'react';
import classNames from 'classnames';

class MyDocument extends Document {
  static async getInitialProps(ctx) {
    const initialProps = await Document.getInitialProps(ctx)
    return { ...initialProps }
  }

  render() {
    return (
      // theme variable to add class dark to html tag conditionally
      <Html className={classNames({'dark': theme === 'dark' })}>
        <Head />
        <body className="antialiased">
          <Main />
          <NextScript />
        </body>
      </Html>
    )
  }
}

export default MyDocument
```
how can I get value of theme in `localStorage`? I've tried to get it in `componentDidmount` but It doesn't work.
how to make it aware of `localStorage` changing?
Could anyone help me? Appreciate!

**Comment by efe-osa** [comment]
@nos-nart You can check this articles https://dev.to/enochndika/dark-mode-with-tailwindcss-in-next-js-2if5 using [next-themes](https://www.npmjs.com/package/next-themes?activeTab=readme) or try out the implementation of the dark mode on [tailwind docs](https://tailwindcss.com/docs/dark-mode)

**Comment by SahasSaurav** [comment]
instead you could use document.documentElement  to target the html element and logically change the class in js.
I have recently implement this in my project
https://github.com/SahasSaurav/next-todos/blob/main/src/reducer/themeReducer.ts
